sdks/python/apache_beam/utils/interactive_utils.py

```diff
@@ -34,6 +38,11 @@ def is_in_ipython():
       is_in_ipython = True
   except ImportError:
     pass  # If dependencies are not available, then not interactive for sure.
+  except (KeyboardInterrupt, SystemExit):
+    raise
+  except:  # pylint: disable=bare-except
+    _LOGGER.info(
+        'Unexpected error occurred, treated as not in IPython.', exc_info=True)
```

Comment:
Optional: This is a case where an explicit `return False` (here and for the ImportError) might make things more clear.

Reply:
Added explicit return statement at each return point.